File: CS/5_seminar/Medium_plus/program.py
from math import sqrt


def is_prime(number):
	prime_nums = [num for num in range(
		1, number + 1) if all(num % val or val == num for val in range(2, int(sqrt(number) + 1)))]
	return number in prime_nums


n = input('Введите натуральное число больше 100: ')
subset = set()
prime_subset = set()
for i in range(0, pow(10, (len(n) - 1))):
	if str(i) in n:
		subset.add(i)
for i in subset:
	if is_prime(i) and i > 1:
		prime_subset.add(i)
print(f'Число {n} имеет {len(subset)} подчисел -> {sorted(subset)}')
print(f'Из них простых {len(prime_subset)} -> {sorted(prime_subset)}')

File: CS/5_seminar/Medium_plus/Program.cs
bool IsPrime(int number)
{
    bool prime = true;
    for (int i = 2; i <= (int)Math.Sqrt(number); i++)
    {
        if (number % i == 0)
        {
            prime = false;
            break;
        }
    }
    return prime;
}

Console.Write("Введите натуральное число больше 100: ");
string n = Console.ReadLine();
SortedSet<int> subset = new SortedSet<int>();
SortedSet<int> prime_subset = new SortedSet<int>();
for (int i = 0; i < (int)Math.Pow(10, n.Length - 1); i++)
{
    if (n.Contains(Convert.ToString(i))) subset.Add(i);
}
Console.WriteLine($"Число {n} имеет {subset.Count} подчисел -> [{String.Join(", ", subset)}]");
foreach (int i in subset)
{
    if (IsPrime(i) && i > 1) prime_subset.Add(i);
}
Console.WriteLine($"Из них простых {prime_subset.Count} -> [{String.Join(", ", prime_subset)}]");